### PR TITLE
Mp Bug Fixes

### LINKED ISSUE #240: Error thrown in MP when TSD open
**Describe the bug**

During multiplayer testing, it was found that the error would throw whenever the TSD was opened. It is unknown whether it is multiplayer-only.

**To Reproduce**

1. Open the TSD page

**Screenshots**

**Environments where the bug occurs**

Delete as appropriate

* Multiplayer multi-crew

**Reference material**

### PR DESCRIPTION
**Pull request by Ollieollieolllie**
Fixes System common vars
Fixes #240

### CODE REVIEW

File: @AH-64D Apache Longbow/Addons/fza_ah64_systems/functions/electrical/fn_electricalBattery.sqf [hunk 1]
```diff
@@ -45,7 +45,7 @@ if (_battBusOn && _acBusOn) then {
 if (_battBusOn && _acBusOn) then {
     _battPower = [_battPower, 1.0, _deltaTime] call BIS_fnc_lerp;
 };
-_heli setVariable ["fza_systems_battPower_pct", _battPower];
+[_heli, "fza_systems_battPower_pct", _battPower] call fza_fnc_updateNetworkGlobal;
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_systems/functions/electrical/fn_electricalBattery.sqf:48]: Isn't this going to setVariable every frame?

File: @AH-64D Apache Longbow/Addons/fza_ah64_ase/functions/fn_rlwr.sqf [hunk 2]
```diff
@@ -71,7 +71,6 @@ if (_rlwrPwr == "on") then {
     } foreach getSensorThreats _heli;
     _rlwrObjects = [_rlwrObjects, [], {_x # 0}, "DESCEND"] call BIS_fnc_sortBy;
 } else { _rlwrObjects = []; };
-_heli setVariable ["fza_ah64_ase_rlwrObjects", _rlwrObjects];
 _rlwrCount = count _rlwrObjects;
 _heli setVariable ["fza_ah64_ase_rlwrCount", _rlwrCount];
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_ase/functions/fn_rlwr.sqf:75]: Shouldn't this also be public according to defs in `eventInit`? If this one was missed there may be others. I would recommend doing a Ctrl+Shift+F of all the variables you are modifying in order to validate all usage sites

File: @AH-64D Apache Longbow/Addons/fza_ah64_common/functions/fn_updateNetworkGlobal.sqf [hunk 3]
```diff
@@ -0,0 +1,27 @@
+/* ----------------------------------------------------------------------------
+Function: fza_fnc_updateNetworkGlobal
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_common/functions/fn_updateNetworkGlobal.sqf:2]: Nice, I like this function. In the future we could even add some profiling here

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/fire/fn_fireHandlepanel.sqf [hunk 4]
```diff
@@ -21,15 +21,15 @@ params ["_heli","_engine","_value"];
 
 switch(_engine) do {
     case "eng1": {
-        _heli setVariable ["fza_ah64_fire1arm", _value, true];
+        [_heli, "fza_ah64_fire1arm", _value] call fza_fnc_updateNetworkGlobal;
         _heli setObjectTextureGlobal ["in_lt_fire1rdy", ["", "\fza_ah64_us\tex\in\pushbut.paa"] select _value];
     };
     case "eng2": {
-        _heli setVariable ["fza_ah64_fire2arm", _value, true];
+        [_heli, "fza_ah64_fire2arm", _value] call fza_fnc_updateNetworkGlobal;
```
[the remark this answers is not in the document]
Comment (mattysmith22) [reply @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/fire/fn_fireHandlepanel.sqf:28]: I see you were just doing it for coonsistency

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/targeting/fn_targetingFCRUpdate.sqf [hunk 5]
```diff
@@ -42,10 +42,10 @@ if (_newNtsIndex == -1) then {
 };
 
 if(_newNtsIndex == -1) then {
-    _heli setVariable ["fza_ah64_fcrNts", [objNull,[0,0,0]], true];
+    [_heli, "fza_ah64_fcrNts", [objNull,[0,0,0]]] call fza_fnc_updateNetworkGlobal;
 } else {
-    _heli setVariable ["fza_ah64_fcrNts", [_fcrTargets # _newNtsIndex # 3,_fcrTargets # _newNtsIndex # 0], true];
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/targeting/fn_targetingFCRUpdate.sqf:47]: This function is run every 2 seconds right? If so these updates are fine

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/targeting/fn_targetingSched.sqf [hunk 6]
```diff
@@ -45,13 +45,13 @@ if (_heli animationPhase "fcr_enable" == 1 && local _heli) then {
                     [_heli] call fza_fnc_targetingFCRUpdate;
                 }
             } else {
-                _heli setVariable ["fza_ah64_fcrState", [FCR_MODE_OFF, time], true];
+                [_heli, "fza_ah64_fcrState", [FCR_MODE_OFF, time]] call fza_fnc_updateNetworkGlobal;
             };
         };
     };
 };
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/targeting/fn_targetingSched.sqf:53]: Is this used now we use ACE laser?

File: @AH-64D Apache Longbow/Addons/fza_ah64_systems/functions/apu/fn_apu.sqf [hunk 7]
```diff
@@ -32,7 +32,7 @@ if (_apuBtnOn && _battBusOn) then {
 } else {
     _apuRPM_pct = [_apuRPM_pct, 0.0, _deltaTime] call BIS_fnc_lerp;
 };
-_heli setVariable ["fza_systems_apuRPM_pct", _apuRPM_pct];
+[_heli, "fza_systems_apuRPM_pct", _apuRPM_pct] call fza_fnc_updateNetworkGlobal;
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_systems/functions/apu/fn_apu.sqf:35]: Will this update every frame?

File: @AH-64D Apache Longbow/Addons/fza_ah64_systems/functions/hydraulics/fn_hydraulicsUtilReservoir.sqf [hunk 8]
```diff
@@ -45,4 +45,4 @@ if (_utilHydLevel_pct < SYS_HYD_MIN_LVL) then {
     //CALL WCA here
 };
 
-_heli setVariable ["fza_systems_utilLevel_pct",  _utilHydLevel_pct];
\ No newline at end of file
+[_heli, "fza_systems_utilLevel_pct", _utilHydLevel_pct] call fza_fnc_updateNetworkGlobal;
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_systems/functions/hydraulics/fn_hydraulicsUtilReservoir.sqf:48]: Will his be updated every frame?

File: @AH-64D Apache Longbow/Addons/fza_ah64_systems/functions/interact/fn_interactBattSwitch.sqf [hunk 9]
```diff
@@ -18,4 +18,4 @@ Author:
 ---------------------------------------------------------------------------- */
 params ["_heli"];
 
-_heli setVariable ["fza_systems_battSwitchOn",  !(_heli getVariable "fza_systems_battSwitchOn")];
+_heli setVariable ["fza_systems_battSwitchOn",  !(_heli getVariable "fza_systems_battSwitchOn"), true];
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_systems/functions/interact/fn_interactBattSwitch.sqf:21]: Should't this use the new function?

File: @AH-64D Apache Longbow/Addons/fza_ah64_systems/functions/hydraulics/fn_hydraulicsPriReservoir.sqf [hunk 10]
```diff
@@ -45,4 +45,4 @@ if (_priHydLevel_pct < SYS_HYD_MIN_LVL) then {
     //CALL WCA here
 };
 
-_heli setVariable ["fza_systems_priLevel_pct",  _priHydLevel_pct];
\ No newline at end of file
+[_heli, "fza_systems_priLevel_pct", _priHydLevel_pct] call fza_fnc_updateNetworkGlobal;
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_systems/functions/hydraulics/fn_hydraulicsPriReservoir.sqf:48]: Will this update every frame?

File: @AH-64D Apache Longbow/Addons/fza_ah64_systems/functions/repair/fn_repair.sqf [hunk 11]
```diff
@@ -48,17 +48,17 @@ _heli setHitPointDamage ["hitengine2",                      0.0];
...
 _heli setHitPointDamage ["hit_hyd_prireservoir",            0.0];
-_heli setVariable ["fza_systems_priLevel_pct",              1.0];
+_heli setVariable ["fza_systems_priLevel_pct",              1.0, true];
 _heli setHitPointDamage ["hit_hyd_priPump",                 0.0];
-_heli setVariable ["fza_systems_utilHydPSI_pct",            1.0];
+_heli setVariable ["fza_systems_utilHydPSI_pct",            1.0, true];
 
 _heli setHitPointDamage ["hit_hyd_utilreservoir",           0.0];
-_heli setVariable ["fza_systems_utilLevel_pct",             1.0];
+_heli setVariable ["fza_systems_utilLevel_pct",             1.0, true];
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_systems/functions/repair/fn_repair.sqf:58]: Shouldn't thee use the new functions?

**Review by mattysmith22 — CHANGES_REQUESTED** [review]
Great start!Just requires some consistency checks and validation that some variables won't update every frame. We should not do per-frame public variables and may need to re-jig what is stored in them to avoid

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/fire/fn_fireHandlepanel.sqf (hunk 4 above)
[the remark this answers is not in the document]
Comment (Ollieollieolllie) [reply @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/fire/fn_fireHandlepanel.sqf:28]: no

File: @AH-64D Apache Longbow/Addons/fza_ah64_systems/functions/electrical/fn_electricalBattery.sqf (hunk 1 above)
Comment (Ollieollieolllie) [reply @AH-64D Apache Longbow/Addons/fza_ah64_systems/functions/electrical/fn_electricalBattery.sqf:48]: only if something changes like system damage, or battery state switch

File: @AH-64D Apache Longbow/Addons/fza_ah64_ase/functions/fn_rlwr.sqf (hunk 2 above)
Comment (Ollieollieolllie) [reply @AH-64D Apache Longbow/Addons/fza_ah64_ase/functions/fn_rlwr.sqf:75]: left it intentionally, one less variable to make public, and each seat is able to access the funcs to calculate identical outputs

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/targeting/fn_targetingSched.sqf (hunk 6 above)
Comment (Ollieollieolllie) [reply @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/targeting/fn_targetingSched.sqf:53]: Currently for constraint box pos

File: @AH-64D Apache Longbow/Addons/fza_ah64_systems/functions/hydraulics/fn_hydraulicsUtilReservoir.sqf (hunk 8 above)
Comment (Ollieollieolllie) [reply @AH-64D Apache Longbow/Addons/fza_ah64_systems/functions/hydraulics/fn_hydraulicsUtilReservoir.sqf:48]: probably

File: @AH-64D Apache Longbow/Addons/fza_ah64_systems/functions/interact/fn_interactBattSwitch.sqf (hunk 9 above)
Comment (Ollieollieolllie) [reply @AH-64D Apache Longbow/Addons/fza_ah64_systems/functions/interact/fn_interactBattSwitch.sqf:21]: the logic flip setting of the variable would make the new func mute, as its always going to be different when set

File: @AH-64D Apache Longbow/Addons/fza_ah64_systems/functions/hydraulics/fn_hydraulicsPriReservoir.sqf (hunk 10 above)
Comment (Ollieollieolllie) [reply @AH-64D Apache Longbow/Addons/fza_ah64_systems/functions/hydraulics/fn_hydraulicsPriReservoir.sqf:48]: probably

Comment (Ollieollieolllie) [reply @AH-64D Apache Longbow/Addons/fza_ah64_systems/functions/hydraulics/fn_hydraulicsPriReservoir.sqf:48]: actually only when system damage occurs

File: @AH-64D Apache Longbow/Addons/fza_ah64_systems/functions/hydraulics/fn_hydraulicsUtilReservoir.sqf (hunk 8 above)
Comment (Ollieollieolllie) [reply @AH-64D Apache Longbow/Addons/fza_ah64_systems/functions/hydraulics/fn_hydraulicsUtilReservoir.sqf:48]: actually only when system damage occurs

File: @AH-64D Apache Longbow/Addons/fza_ah64_systems/functions/repair/fn_repair.sqf (hunk 11 above)
Comment (Ollieollieolllie) [reply @AH-64D Apache Longbow/Addons/fza_ah64_systems/functions/repair/fn_repair.sqf:58]: possibly, but its ran Infrequently

**Pull request closed without merging**